ProcessLib/Process.h

```diff
@@ -259,32 +260,34 @@ class Process
 	{
 		DBUG("Process output.");
 
-		std::string const property_name = "Result";
-
-		// Get or create a property vector for results.
-		boost::optional<MeshLib::PropertyVector<double>&> result;
-		if (_mesh.getProperties().hasPropertyVector(property_name))
-		{
-			result = _mesh.getProperties().template getPropertyVector<double>(
-			    property_name);
-		}
-		else
-		{
-			result =
-			    _mesh.getProperties().template createNewPropertyVector<double>(
-			        property_name, MeshLib::MeshItemType::Node);
+		// Copy result
 #ifdef USE_PETSC
-			result->resize(_x->getLocalSize() + _x->getGhostSize());
+		std::vector<double> x_copy(_x->getLocalSize() + _x->getGhostSize());
 #else
-			result->resize(_x->size());
+		std::vector<double> x_copy(_x->size());
 #endif
-		}
-
-		assert(result);
-
-		// Copy result
-		_x->copyValues(*result);
+		_x->copyValues(x_copy);
 
+		std::size_t const n_nodes = _mesh.getNNodes();
+		for (ProcessVariable& pv : _process_variables)
+		{
+			auto& output_data = pv.getOrCreateMeshProperty();
+
+			int const n_components = pv.getNComponents();
+			for (std::size_t node_id = 0; node_id < n_nodes; ++node_id)
+			{
+				MeshLib::Location const l(_mesh.getID(),
+				                          MeshLib::MeshItemType::Node, node_id);
+				for (int component_id = 0; component_id < n_components;
+				     ++component_id)
+				{
+					auto const global_index =
```

Comment:
Yes, using the global index is wrong inside node and component loops. The global index is for the index of any entry of the global matrix and global vector, and it already contains the component id. Here a function  to fetch the value for this node with a known of global id and a known of component id is needed.